### PR TITLE
Steps 1-4 of the file processing Code

### PR DESCRIPTION
**Pull request by Shashank411**

### CODE REVIEW

File: kairon/data_processor/processor.py [hunk 1]
```diff
@@ -2364,3 +2364,27 @@ def __get_all_history(bot: Text):
             item.pop("bot")
             item["_id"] = item["_id"].__str__()
             yield item
+
+    @staticmethod
+    def is_daily_file_limit_exceeded(bot: Text, raise_exception=True):
```
Comment (sfahad1414) [inline kairon/data_processor/processor.py:2370]: change name from is_daily_file_limit_exceeded to check_data_generation_limit

File: tests/testing_data/system.yaml [hunk 2]
```diff
@@ -23,5 +23,6 @@ action:
   url: ${ACTION_SERVER_URL:"http://localhost:5055/webhook"}
 
 knowledge_graph:
```
Comment (sfahad1414) [inline tests/testing_data/system.yaml:25]: change knowledge_graph to data_generation

File: kairon/api/app/routers/bot.py [hunk 3]
```diff
@@ -606,3 +628,14 @@ async def update_training_data_generator_status(
     TrainingDataGenerationProcessor.retreive_response_and_set_status(request_data, current_user.get_bot(),
                                                                      current_user.get_user())
     return {"message": "Status updated successfully!"}
+
+
+@router.get("/train_data/history", response_model=Response)
```
Comment (sfahad1414) [inline kairon/api/app/routers/bot.py:639]: rename route /traind_data to data/generation/history

File: kairon/__main__.py [hunk 4]
```diff
@@ -21,8 +33,12 @@ def main():
     connect(host=Utility.environment['database']['url'])
     logger.info(arguments.bot)
     logger.info(arguments.user)
-    start_training(arguments.bot, arguments.user, arguments.token)
+    if arguments.which == 'training_parser':
```
Comment (sfahad1414) [inline kairon/__main__.py:36]: 1. change from training_parser to training
2. also need to check model trianing

File: kairon/__main__.py [hunk 5]
```diff
@@ -21,8 +33,12 @@ def main():
...
+        logger.info(arguments.token)
+        start_training(arguments.bot, arguments.user, arguments.token)
+    elif arguments.which == 'kb_parser':
```
Comment (sfahad1414) [inline kairon/__main__.py:39]: change kb_parser to data_generation

**Review by sfahad1414 — CHANGES_REQUESTED** [review]

File: kairon/api/app/routers/bot.py [hunk 6]
```diff
@@ -283,7 +283,8 @@ async def chat(
     if Utility.environment.get('model') and Utility.environment['model']['train'].get('agent_url'):
         agent_url = Utility.environment['model']['train'].get('agent_url')
         token = auth.create_access_token(data={"sub": current_user.email})
-        response = Utility.http_request('post', urljoin(agent_url, "/api/bot/chat"), token.decode('utf8'), current_user.get_user(), json={'data': request_data.data})
+        response = Utility.http_request('post', urljoin(agent_url, "/api/bot/chat"), token.decode('utf8'),
```
Comment (sfahad1414) [inline kairon/api/app/routers/bot.py:287]: why this line was touch as there is no changes

File: kairon/api/app/routers/bot.py [hunk 7]
```diff
@@ -432,7 +433,7 @@ async def download_model(
 
 
 @router.get("/endpoint", response_model=Response)
-async def get_endpoint(current_user: User = Depends(auth.get_current_user),):
+async def get_endpoint(current_user: User = Depends(auth.get_current_user), ):
```
Comment (sfahad1414) [inline kairon/api/app/routers/bot.py:458]: no changes here as well

File: kairon/api/app/routers/bot.py [hunk 8]
```diff
@@ -543,8 +544,8 @@ async def get_http_action(action: str = Path(default=None, description="action n
     Returns configuration set for the HTTP action
     """
     http_action_config = mongo_processor.get_http_action_config(action_name=action,
-                                                                           user=current_user.get_user(),
-                                                                           bot=current_user.bot)
+                                                                user=current_user.get_user(),
+                                                                bot=current_user.bot)
```
Comment (sfahad1414) [inline kairon/api/app/routers/bot.py:570]: no changes here as well revert

File: kairon/api/app/routers/bot.py [hunk 9]
```diff
@@ -577,3 +578,31 @@ async def delete_http_action(action: str = Path(default=None, description="actio
...
+    status = mongo_processor.add_training_data(
+        training_data=request_data.training_data,
+        bot=current_user.get_bot(),
+        user=current_user.get_user(),
+        is_integration=current_user.get_integration_status()
+    )
+    return {"message": "Training data added successfully!", "data": status}
+
+
+@router.put("/processing-status", response_model=Response)
```
Comment (sfahad1414) [inline kairon/api/app/routers/bot.py:624]: rename this from processing-status

File: kairon/api/app/routers/bot.py [hunk 10]
```diff
@@ -577,3 +578,31 @@ async def delete_http_action(action: str = Path(default=None, description="actio
         raise AppException(e)
     message = "HTTP action deleted"
     return Response(message=message)
+
+
+@router.post("/training-data", response_model=Response)
```
Comment (sfahad1414) [inline kairon/api/app/routers/bot.py:605]: rename this from training-data to training_examples/bulk

File: kairon/api/models.py [hunk 11]
```diff
@@ -105,8 +108,8 @@ def validate_password(cls, v, values, **kwargs):
     @validator("confirm_password")
     def validate_confirm_password(cls, v, values, **kwargs):
         if (
-            "password" in values
-            and v.get_secret_value() != values["password"].get_secret_value()
+                "password" in values
```
Comment (sfahad1414) [inline kairon/api/models.py:106]: revert this code as there is no changes

**Review by sfahad1414 — CHANGES_REQUESTED** [review]

File: kairon/api/app/routers/bot.py (hunk 10 above)
Comment (udit-pandey1) [reply kairon/api/app/routers/bot.py:605]: /data/bulk

**Pull request closed without merging**